core/src/main/java/org/ehcache/StandaloneCacheBuilder.java

```diff
@@ -61,16 +59,13 @@ public StandaloneCacheBuilder(final Class<K> keyType, final Class<V> valueType,
 
   T build(ServiceLocator serviceLocator) {
     Store.Provider storeProvider = serviceLocator.findService(Store.Provider.class);
-    if (serializationProvider == null) {
-      this.serializationProvider = serviceLocator.findService(SerializationProvider.class);
-    }
 
     final StoreConfigurationImpl<K, V> storeConfig = new StoreConfigurationImpl<K, V>(keyType, valueType,
-        capacityConstraint, evictionVeto, evictionPrioritizer, classLoader, expiry, serializationProvider);
-    final Store<K, V> store = storeProvider.createStore(storeConfig);
+        capacityConstraint, evictionVeto, evictionPrioritizer, classLoader, expiry);
+    final Store<K, V> store = storeProvider.createStore(serviceLocator, storeConfig);
```

Comment:
There should not be a need to pass in the serviceLocator here. Upon starting a service receives the `ServiceProvider` it is supposed to use.